lib/ae_mdw/db/sync/contract.ex

```diff
@@ -139,62 +144,44 @@ defmodule AeMdw.Db.Sync.Contract do
-      _int_call -> false
-    end)
-    |> Enum.map(fn
-      {{:internal_call_tx, "Oracle.extend"}, %{info: aetx}} ->
-        {:oracle_extend_tx, tx} = :aetx.specialize_type(aetx)
-        Oracle.extend_mutation(tx, block_index, call_txi)
+  defp oracle_and_name_mutations(int_calls, {height, _mbi} = block_index, call_txi) do
+    Enum.map(int_calls, fn
+      {local_idx, "Oracle.extend", :oracle_extend_tx, _aetx, tx, _tx_hash} ->
+        Oracle.extend_mutation(tx, block_index, {call_txi, local_idx})
 
-      {{:internal_call_tx, "Oracle.register"}, %{info: aetx, tx_hash: tx_hash}} ->
-        {:oracle_register_tx, tx} = :aetx.specialize_type(aetx)
-        Oracle.register_mutations(tx, tx_hash, block_index, call_txi)
+      {local_idx, "Oracle.register", :oracle_register_tx, _aetx, tx, tx_hash} ->
+        Oracle.register_mutations(tx, tx_hash, block_index, {call_txi, local_idx})
 
-      {{:internal_call_tx, "Oracle.respond"}, %{info: aetx}} ->
-        {:oracle_response_tx, tx} = :aetx.specialize_type(aetx)
+      {_local_idx, "Oracle.respond", :oracle_respond_tx, _aetx, tx, _tx_hash} ->
         Oracle.response_mutation(tx, block_index, call_txi)
 
-      {{:internal_call_tx, "Oracle.query"}, %{info: aetx}} ->
-        {:oracle_query_tx, tx} = :aetx.specialize_type(aetx)
+      {_local_idx, "Oracle.query", :oracle_query_tx, _aetx, tx, _tx_hash} ->
         Oracle.query_mutation(tx, height)
 
-      {{:internal_call_tx, "AENS.claim"}, %{info: aetx, tx_hash: tx_hash}} ->
-        {:name_claim_tx, tx} = :aetx.specialize_type(aetx)
-        Name.name_claim_mutations(tx, tx_hash, block_index, call_txi)
+      {local_idx, "AENS.claim", :name_claim_tx, _aetx, tx, tx_hash} ->
+        Name.name_claim_mutations(tx, tx_hash, block_index, {call_txi, local_idx})
 
-      {{:internal_call_tx, "AENS.update"}, %{info: aetx}} ->
-        {:name_update_tx, tx} = :aetx.specialize_type(aetx)
-        Name.update_mutations(tx, call_txi, block_index, true)
+      {local_idx, "AENS.update", :name_update_tx, _aetx, tx, _tx_hash} ->
+        Name.update_mutations(tx, {call_txi, local_idx}, block_index, true)
 
-      {{:internal_call_tx, "AENS.transfer"}, %{info: aetx}} ->
-        {:name_transfer_tx, tx} = :aetx.specialize_type(aetx)
-        NameTransferMutation.new(tx, call_txi, block_index)
-
-      {{:internal_call_tx, "AENS.revoke"}, %{info: aetx}} ->
-        {:name_revoke_tx, tx} = :aetx.specialize_type(aetx)
+      {local_idx, "AENS.transfer", :name_transfer_tx, _aetx, tx, _tx_hash} ->
+        NameTransferMutation.new(tx, {call_txi, local_idx}, block_index)
 
+      {local_idx, "AENS.revoke", :name_revoke_tx, _aetx, tx, _tx_hash} ->
         tx
         |> :aens_revoke_tx.name_hash()
-        |> NameRevokeMutation.new(call_txi, block_index)
+        |> NameRevokeMutation.new({call_txi, local_idx}, block_index)
+
+      {_local_idx, fname, _tx_type, _aetx, _tx, _tx_hash} when fname in @ignored_tx_calls ->
```

Comment:
I would agree to make a full match to any other event like before in such a way that I would care only about the events that need to be handled (the ignored ones would be the complement set that might change). An pattern match error would be caught by a sync but would save me some time detecting which new one should match.

Reply:
Pattern-matching error display the structure that it wasn't pattern-matched to. added a `raise` anyway

Reply:
It was matching any other event before:
https://github.com/aeternity/ae_mdw/blob/bc54cd63044bb88906637aeb0445c4ea4e132d5c/lib/ae_mdw/db/sync/contract.ex#L153
To ignore internal calls, we would need just like that, for any other that doesn't have to be handled.

Reply:
I'd rather raise an exception, if we mistakenly add a clause that doesn't match the `tx_type` (e.g. `{local_idx, "ANES.revoke", :name_foo_tx, ...}` we would never find out about it

Reply:
`fname=AENS.revoke` wouldn't match either for `fname in @ignored_tx_calls` but it's okay, as new events are introduced some tests would be able to cover it.